js/src/traceable.ts

```diff
@@ -167,6 +180,184 @@ const getTracingRunTree = (
   return runTree;
 };
 
+// idea: store the state of the promise outside
+// but only when the promise is "consumed"
+const getSerializablePromise = <T = unknown>(arg: Promise<T>) => {
+  const proxyState: {
+    current: ["resolve", unknown] | ["reject", unknown] | undefined;
+  } = { current: undefined };
+
+  const promiseProxy = new Proxy(arg, {
+    get(target, prop, receiver) {
+      if (prop === "then") {
+        const boundThen = arg[prop].bind(arg);
+        return (
+          resolve: (value: unknown) => unknown,
+          reject?: (error: unknown) => unknown
+        ) => {
+          return boundThen(
+            (value) => {
+              proxyState.current = ["resolve", value];
+              return resolve(value);
+            },
+            (error) => {
```

Comment:
<img width="1840" alt="Screenshot 2024-05-06 at 3 02 08 PM" src="https://github.com/langchain-ai/langsmith-sdk/assets/56902/de8d36c1-ce07-466b-8d7c-52505d4e31e1">

I think this results in swallowing promise rejections when users call `.then` without 2nd arg